packages/api/src/middlewares/withTelemetry.ts

```diff
@@ -1,43 +1,98 @@
-import type { api } from "@blobscan/open-telemetry";
+import { api } from "@blobscan/open-telemetry";
 
-import { tracer } from "../instrumentation";
+import { meter, tracer } from "../instrumentation";
 import { t } from "../trpc-client";
 
+type APICounter = {
+  scope?: string;
+};
+
+type RequestsTotalCounter = APICounter & {
+  endpoint: string;
+  status_code: string;
+  method: string;
+};
+
+const apiRequestsTotalCounter = meter.createCounter<RequestsTotalCounter>(
+  "blobscan_api_requests_total",
+  {
+    valueType: api.ValueType.INT,
+    description:
+      "Number of requests made to Blobscan API by endpoint, status_code and method",
+  }
+);
+
+const apiRequestsDurationMsHistogram = meter.createHistogram<APICounter>(
+  "blobscan_api_requests_duration_ms",
+  {
+    description: "Duration of all the requests made to Blobscan API",
+    valueType: api.ValueType.INT,
+    unit: "ms",
+  }
+);
+
+function getEndpoint(url: string) {
+  return url.split("?")[0] ?? "";
```

Comment:
Something like [this](https://stackoverflow.com/a/24006120) looks better to me

Reply:
Sure